vibrato/src/state.rs

```diff
@@ -0,0 +1,101 @@
+//! Maintainer of an input sentence and tokenized results.
+use crate::dictionary::mapper::{ConnIdCounter, ConnIdProbs};
+use crate::dictionary::Dictionary;
+use crate::errors::Result;
+use crate::sentence::Sentence;
+use crate::token::{Token, TokenIter};
+use crate::tokenizer::lattice::{Lattice, Node};
+
+/// Maintainer of an input sentence and tokenized results.
+///
+/// It also holds the internal data structures used in tokenization,
+/// which can be reused to avoid unnecessary memory reallocation.
+pub struct State<'a> {
```

Comment:
Shouldn't this struct be named `Lattice`?
The name `State` is difficult to understand for those who do not know the internal implementation.

Reply:
@vbkaisetsu How about `Worker`?

I think the role of this structure is to provide a routine for tokenization. For many light users, the internal data structure will not be interesting, so `Worker` is more friendly than the technical name `Lattice`.

e.g.,

```rust
let tokenizer = vibrato::Tokenizer::new(dict);
let mut worker = tokenizer.new_worker();
worker.reset_sentence("京都東京都");
worker.tokenize();
worker.token(0);
...
```

Reply:
ok

Reply:
https://github.com/daac-tools/vibrato/pull/21/commits/c69049e2cd381f14cd295ba64c61fd3e97b143c2